### PR TITLE
T42: Edit profile endpoints

### LINKED ISSUE #42: API endpoints to edit profile
The API endpoints for users to edit their profile information

### PR DESCRIPTION
**Pull request by Dennull**
Closes #42. An API endpoint was added for the edit profile modal. The new endpoint is a PUT endpoint that allows users to modify these profile details:

- Bio
- Birthday
- Display name

### CODE REVIEW

File: models/ProfileModel.js [hunk 1]
```diff
@@ -1,11 +1,21 @@
-const getUserPosts = `WITH post_likes AS (
-    SELECT "postId", 
+const editProfile = `
+  UPDATE app_user
+  SET "displayName" = $1,"birthDate" = $2, "bio" = $3
+  WHERE "userId" = $4
+  RETURNING "displayName", "birthDate", bio;
+`;
```
Comment (Dennull) [inline models/ProfileModel.js]: This is the only new query, every other query change was just formatting

File: models/ProfileModel.js [hunk 2]
```diff
@@ -1,11 +1,21 @@
-const getUserPosts = `WITH post_likes AS (
-    SELECT "postId", 
+const editProfile = `
+  UPDATE app_user
+  SET "displayName" = $1,"birthDate" = $2, "bio" = $3
+  WHERE "userId" = $4
+  RETURNING "displayName", "birthDate", bio;
```
Comment (NasihNazeem) [inline models/ProfileModel.js]: Query seems concise enough, I think `bio` at the `RETURNING` statement needs to be `"bio"`

File: controllers/ProfileController.js [hunk 3]
```diff
@@ -1,6 +1,22 @@
 const pool = require("../database/db");
 const profileQueries = require("../models/ProfileModel");
 
+const editProfile = async (req, res) => {
+  try {
+    const { displayName, birthDate, bio, userId } = req.body;
```
Comment (NasihNazeem) [inline controllers/ProfileController.js:6]: should we provide some order to these variables? like `userId` to keep it consistent, and then the rest?

**Review by NasihNazeem — CHANGES_REQUESTED** [review]
Not much, just some nitpick comments or suggestions that we can use moving forward. 

I think it'd be cool to open another ticket for query formatting at a later date.

File: models/ProfileModel.js (hunk 2 above)
Comment (Dennull) [reply models/ProfileModel.js]: I actually ended up removing this because we can just use the data from the input fields to update the client side

**Review by NasihNazeem — APPROVED** [review]

**Pull request merged**